docs/localdev.md

```diff
@@ -0,0 +1,89 @@
+1. If you haven't done so create a fork of the openshift/external-dns-operator to your github account
+2. Clone the repo from your github account to your local machine
+3. Change directory `cd <projects-dir>/external-dns-operator`
+4. Ensure your can access your CRC environment
+   ```bash
+   $ export KUBECONFIG=~/.crc/machines/crc/kubeconfig
+   ```
+5. Access to the CRC internal registry
+   ```bash
+   # assume that podman (tool used to create and maintain containers) is our default
+   # assume our default user is kubeadmin
+   $ podman login -u kubeadmin -p $(oc whoami -t) default-route-openshift-image-registry.apps-crc.testing --tls-verify=false
+   ```
+6. We can now follow the standard make recipes
+   ```bash
+   # url format is default-route-openshift-image-registry.apps-crc.testing/<namespace>/<image-name>:tag
+   $ export IMG=default-route-openshift-image-registry.apps-crc.testing/external-dns-operator/external-dns-operator:dev
+   $ CONTAINER_ENGINE=podman make image-build
+   $ TLS_VERFIY=--tls-verify=false make image-push
+   # check that the image was pushed
+   $ oc -n external-dns-operator get is
+   ``` 
+7. Now you can deploy an instance of ExternalDNS
+   ```bash
+   # switch to the external-dns-namespace
+   $ oc project external-dns-operator
+   $ make deploy 
+   # this deploys the operator in CRC
+   # check deployment
+   $ oc get all
+   # if there are errors/problems
+   $ oc get events
+   ```
+8. Run the following command to create the credentials secret on AWS.\
+   Ensure you have your AWS credentials set up in `~/.aws/credentials` if not please look at the onbording document to find out how to get them.
+   ```bash
+   $ oc create secret generic aws-access-key \
+   --from-literal=aws_access_key_id=${ACCESS_KEY_ID} \
+   --from-literal=aws_secret_access_key=${ACCESS_SECRET_KEY}
+   ```
+   *Note*: other provider options can be found in `api/v1alpha1/externaldns_types.go`, e.g. the `ExternalDNSAWSProviderOptions` structure for AWS.
+
+   Execute the following
+   ```bash
+   $ oc apply -k config/samples/aws
+   # check for the newly created ExternalDNS
+   $ oc get externaldns sample -o yaml  
+   ``` 
+   *Note*: other providers available in `config/samples/`
+
+### Running the external-dns-operator locally (with CRC)
+
+1. Assume that the operator hase been deployed (steps 6 in the preveious section)
+2. Execute the following
+   ```bash
+   # scale the cluster version operator to 0
+   $ oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
+   # scale the current deployed external-dns-operator to 0
+   $ oc scale --replicas 0 -n external-dns-operator deployments external-dns-operator
+   $ make run
+   ```
+3. Once you are completed with debug/testing/development remember to scale the cluster-version-operator to 1 if you want to continue with other operations.
+
+### Working with cluster-bot
+
+Ensure that the KUBECONFIG environment variable points to the cluster-bot config \
```

Comment:
The cluster bot is too specific to Red Hat, not sure whether it makes sense to add it here as we try to target the upstream community. Also, it creates a distant (not local) cluster.

May be this chapter can be generalized to something like:
```
If you have any other cluster you'd like to test the operator on you just need to set KUBECONFIG and follow the instructions from X, Y, Z.
```
WDYT?